include/output.hpp

```diff
@@ -247,6 +248,16 @@ namespace mamba
         ProgressProxy add_progress_bar(const std::string& name);
         void reset_multi_progress();
 
+        nlohmann::json json_log;
+        void json_write(const std::string& key, bool value);
+        void json_write(const std::string& key, const std::string& value);
+        void json_write(const nlohmann::json& j);
+        void json_append(const std::string& value);
+        void json_append(const nlohmann::json& j);
+        void json_down(const std::string& key);
+        void json_up();
+        void json_print();
```

Comment:
I wonder if this API should not be in a dedicated class (JSON builder / formatter or whatever) and call `Console::print()` when the json is finished. This violates the principle of single responsibility.

Reply:
Agreed!